src/game/Spell.cpp

```diff
@@ -5056,11 +5056,18 @@ SpellCastResult Spell::CheckCast(bool strict)
             {
                 Creature* pet = m_caster->GetPet();
                 if (!pet)
-                    return SPELL_FAILED_NO_PET;
-
-                if (pet->isAlive())
-                    return SPELL_FAILED_ALREADY_HAVE_SUMMON;
-
+                {
+                    Pet* dbPet = new Pet;
+                    SpellCastResult result = dbPet->TryLoadFromDB((Player*)m_caster, 0);
```

Comment:
unless anything except hunter players are supposed to know 982 (Revive Pet) then I think we're safe.